Source/Core/Core/HW/DSPHLE/DSPHLE.cpp

```diff
@@ -79,9 +79,10 @@ void DSPHLE::SwapUCode(u32 crc)
 {
   m_mail_handler.Clear();
 
-  if (m_last_ucode == nullptr)
+  if (!m_last_ucode || UCodeInterface::GetCRC(m_last_ucode.get()) != crc)
```

Comment:
For clarity, maybe invert the conditional to avoid the extra negations
```suggestion
  if (m_last_ucode && UCodeInterface::GetCRC(m_last_ucode.get()) == crc)
  {
    m_ucode = std::move(m_last_ucode);
  }
  ...
```